doc/patch-author-guide.md

```diff
@@ -113,8 +113,15 @@ patch module is loaded.
 
 `kpatch-macros.h` provides `KPATCH_LOAD_HOOK` and `KPATCH_UNLOAD_HOOK` macros
 to define such functions.  The signature of both hook functions is `void
-foo(void)` and and they may run in `stop_machine` context (so they must not
-sleep).
+foo(void)`.  Their execution context is as follows:
+
+* For objects that are already loaded, hook functions will be run by
+`stop_machine` as part of applying or removing a patch. (Therefore the hooks
+must not block or sleep.)
+
+* For objects not yet loaded, a module-notifier will execute load hooks when
```

Comment:
Similar change here: "For patches to kernel modules which haven't been loaded yet..."